### PR TITLE
Remove ObjectStore and  finish the implementation of EnvS3::new_writable_file

### PR DESCRIPTION
**Pull request by sduzh**
## Problem Summary(Required) ：

Remove ObjectStore and finish the implementation of EnvS3::new_writable_file

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

### CODE REVIEW

File: be/src/common/s3_uri.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <string>
+#include <string_view>
+
+namespace starrocks {
+
+class S3URI {
```
Comment (starrocks-xueli) [inline be/src/common/s3_uri.h:10]: better to give a correct example here.

File: be/src/env/env_s3.cpp [hunk 2]
```diff
@@ -66,33 +82,96 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const
...
+    if (!access_key_id.empty() && !secret_access_key.empty()) {
+        auto credentials = std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(access_key_id, secret_access_key);
+        return std::make_shared<Aws::S3::S3Client>(credentials, config);
+    } else {
+        // if not cred provided, we can use default cred in aws profile.
+        return std::make_shared<Aws::S3::S3Client>(config);
+    }
+}
+
+StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const RandomAccessFileOptions& opts,
```
Comment (starrocks-xueli) [inline be/src/env/env_s3.cpp:146]: i don't think it's necessary to create a new client for every file.

Comment (sduzh) [reply be/src/env/env_s3.cpp:146]: I already added a TODO above.

**Review by dirtysalt — APPROVED** [review]

**Review by wuleistarrocks — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

File: be/src/env/env_s3.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/env/env_s3.cpp:146]: https://github.com/StarRocks/starrocks/issues/3794

File: be/src/common/s3_uri.h (hunk 1 above)
Comment (sduzh) [reply be/src/common/s3_uri.h:10]: https://github.com/StarRocks/starrocks/issues/3793

**Pull request merged**